msbuild/rest-api-client-generation/README.md

```diff
@@ -0,0 +1,272 @@
+
+  protected override string GenerateFullPathToTool()
+  {
+      return $"{NSwagCommandFullPath}\\NSwag.exe";
+  }
+  ```
+
+- There are many method that we can override. At least we will need to work on two for the current implementation
+
+  01. Defining the command parameter
+
+      ```csharp
+      protected override string GenerateCommandLineCommands()
+      {
+          return $"openapi2csclient /input:{InputOpenApiSpec}  /classname:{ClientClassName} /namespace:{ClientNamespaceName} /output:{FolderClientClass}\\{ClientClassName}.cs";
+      }
+      ```
+
+  01. Parameters Validation
+
+      ```csharp
+      protected override bool ValidateParameters()
+      {
+            //http address is not allowed
+            var valid = true;
+            if (InputOpenApiSpec.StartsWith("http:") || InputOpenApiSpec.StartsWith("https:"))
+            {
+                valid = false;
+                Log.LogError("URL is not allowed");
+            }
+  
+            return valid;
+      }
+      ```
+
+      _Note:_This simple validation could be done in other way on the MSBuild file, but it is recommended do it in C# code and encapsulate the command and the logic.
+
+- Build the project, everything should compile.
+- Create a console app to use our new MSBuild Task
+
+  - Create a Console App, we will call "PetReaderToolTaskConsoleApp". We are going to use net6.0 in our case. Mark it as startup project
+  - Create a Library project to generate the code, called "PetRestApiClient". NetStandard2.1
+  - On "PetReaderToolTaskConsoleApp" create dependency to "PetRestApiClient"
+  - On PetRestApiClient project create a folder "PetRestApiClient", this folder will contain the generated code and delete Class1.cs automatically generated.
+  - On PetRestApiClient add the following NuGet packages:
+    - Newtonsoft.Json, it will be needed to compile the generated client
+    - System.ComponentModel.Annotations, it will be needed to compile the generated client
+- On PetRestApiClient, create a text file named petshop-openapi-spec.json (on root). We are going to add the OpenApi spec, please copy the content from [here](https://petstore.swagger.io/v2/swagger.json) inside the file. We like repetitive build and depending only from the input, consuming directly the api could transform a build which works today to a build which fails tomorrow from the same source. In this example, we are going to raise a build error if you choose a URL as OpenApi spec input.
+- :warning: A general rebuild won't work. You will see errors like 'unable to copy or delete RestApiClientGenerator.dll'. This is because we are trying to build the MBuild Custom Task on the same build process which use it. Select "PetReaderToolTaskConsoleApp" and rebuild only that project. The another solution is put the Custom Task in a complete independent Visual Studio Solutions as we did on [Custom task code generation](../custom-task-code-generation/) example. We are testing different flavors.
```

Comment:
```suggestion
- :warning: A general rebuild won't work. You will see errors like 'unable to copy or delete RestApiClientGenerator.dll'. This is because we are trying to build the MBuild Custom Task on the same build process which use it. Select "PetReaderToolTaskConsoleApp" and rebuild only that project. The another solution is put the Custom Task in a completely independent Visual Studio Solutions as we did on [Custom task code generation](../custom-task-code-generation/) example. We are testing different flavors.
```